src/Geometry/BoundingBox.h

```diff
@@ -1,43 +1,187 @@
 #pragma once
 
+// Local headers
 #include "Vector2.h"
 #include "Size.h"
 
+// Standard library headers
+#include <cmath>
+
 namespace Geometry {
-	template<class T>
-	struct BoundingBox {
-		Vector2<T> start;
-		Vector2<T> end;
-
-		T width() const;
-
-		T height() const;
-
-		Size<T> size() const;
-
-		bool contains(const Vector2<T> &point) const;
-	};
-
-	template<class T>
-	T BoundingBox<T>::width() const {
-		return end.x - start.x;
-	}
-
-	template<class T>
-	T BoundingBox<T>::height() const {
-		return end.y - start.y;
-	}
-
-	template<class T>
-	bool BoundingBox<T>::contains(const Vector2<T> &point) const {
-		return point.x >= start.x
-		       && point.x < end.x
-		       && point.y >= start.y
-		       && point.y < end.y;
-	}
-
-	template<class T>
-	Size<T> BoundingBox<T>::size() const {
-		return Size<T>(this->width(), this->height());
-	}
+  /// General class for representing a 2D maximum point
+  template<typename T>
+  class Maximum2 : public Vector2<T>
+  {
+  public:
+    /// Default constructor initialises nothing
+    Maximum2() = default;
+
+    /// Construct with the values
+    /// \param x X coordinate
+    /// \param y Y coordinate
+    Maximum2(T x, T y)
+      : Vector2<T>(x, y)
+    {
+    }
+
+    /// Adjust so it is the maximum of this and the other point
+    /// \param p Point
+    /// \param offset Value to add to component of p
+    void add(const Vector2<T>& p, T offset)
```

Comment:
I think add is the wrong name here.  Something more like "expand" or "include" as we're expanding the geometry to include a point. in its bounding box.

Reply:
I prevaricated over a while as add seems to imply we are building a set of points rather than altering the bounds. Changed to expand even though the point added will not necessarily expand the bounds. Couldn't think of an appropriate word but could have written a sentence in snake case which would have been rather pointless.